Updated composer.json

phansys opened the pull request:
* Fixed indentation to 4 spaces
* Removed "version" (https://getcomposer.org/doc/04-schema.md#version)
* Added "branch-alias" in order to allow semver

J7mbo commented:
Unfortunately this can't currently be merged - could you update to work with the latest version?

phansys commented:
Rebased.
Note that I've updated the requirement for ```"phpunit/phpunit``` to ```~4.5,>=4.5.1``` in order to allow greater versions from 4.x branch but starting from your current restrictive ```4.5.1``` constraint.

J7mbo commented:
Seeing as we're now on `1.0.2`, does having `"dev-master": "1.0-dev"` matter here? Just wondering as I haven't used branch aliases before :-)

phansys commented:
IMO, since your ```master``` branch tracks the latest dev updates for ```1.x``` branch, it is a good idea to give the developers the ability to install these updates in a sem-ver basis: https://getcomposer.org/doc/articles/aliases.md

Danack commented on composer.json:
```diff
@@ -3,11 +3,10 @@
     },
     "require-dev": {
-        "phpunit/phpunit": "4.5.1"
+        "phpunit/phpunit": "~4.5,>=4.5.1"
```
Using `^4.51` would be appropriate I think - https://getcomposer.org/doc/01-basic-usage.md#next-significant-release-tilde-and-caret-operators-

phansys replied:
Yeah @Danack, the caret operator has the same result if you use it as ```^4.5.1```. The only reason I usually suggest the tilde operator is because some people yet use older composer versions (where caret isn't available).
